lib/app/features/feed/create_article/providers/create_article_notifier.c.dart

```diff
@@ -0,0 +1,115 @@
+import 'package:ion/app/features/nostr/providers/nostr_upload_notifier.c.dart';
+import 'package:ion/app/services/compressor/compress_service.c.dart';
+import 'package:ion/app/services/media_service/media_service.c.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'create_article_notifier.c.g.dart';
+
+@Riverpod(dependencies: [])
+class CreateArticleNotifier extends _$CreateArticleNotifier {
+  @override
+  FutureOr<void> build() {}
+
+  Future<void> create({
+    required String content,
+    String? title,
+    String? summary,
+    String? imageId,
+    DateTime? publishedAt,
+    List<String>? mediaIds,
+  }) async {
+    state = const AsyncValue.loading();
+
+    state = await AsyncValue.guard(() async {
+      final files = <FileMetadata>[];
+      final uploadedUrls = <String, String>{};
+      final mediaAttachments = <MediaAttachment>[];
+
+      if (mediaIds != null && mediaIds.isNotEmpty) {
+        await Future.wait(
+          mediaIds.map((id) async {
+            final (:fileMetadata, :mediaAttachment) = await _uploadImage(id);
+            uploadedUrls[id] = mediaAttachment.url;
+            files.add(fileMetadata);
+            mediaAttachments.add(mediaAttachment);
+          }),
+        );
+        content = _replaceImagePathsWithUrls(content, uploadedUrls);
+      }
+
+      String? imageUrl;
+      if (imageId != null) {
+        final uploadResult = await _uploadImage(imageId);
+        imageUrl = uploadResult.mediaAttachment.url;
+      }
+
+      final relatedHashtags = ArticleData.extractHashtagsFromMarkdown(content);
+
+      final articleData = ArticleData(
+        title: title,
+        summary: summary,
+        image: imageUrl,
+        content: content,
+        media: {for (final attachment in mediaAttachments) attachment.url: attachment},
+        relatedHashtags: relatedHashtags,
+        publishedAt: publishedAt ?? DateTime.now(),
+      );
+
+      await ref.read(nostrNotifierProvider.notifier).sendEntitiesData([...files, articleData]);
+    });
+  }
+
+  String _replaceImagePathsWithUrls(String content, Map<String, String> uploadedUrls) {
+    final parsedContent = jsonDecode(content) as List<dynamic>;
+
+    for (final operation in parsedContent) {
+      if (operation is Map<String, dynamic> &&
+          operation.containsKey('insert') &&
+          operation['insert'] is Map<String, dynamic>) {
+        final insertData = operation['insert'] as Map<String, dynamic>;
+
+        if (insertData.containsKey(textEditorSingleImageKey)) {
+          final localPath = insertData[textEditorSingleImageKey] as String?;
+          if (localPath != null && uploadedUrls.containsKey(localPath)) {
+            insertData[textEditorSingleImageKey] = uploadedUrls[localPath];
+          }
+        }
+      }
+    }
+
+    return jsonEncode(parsedContent);
+  }
+
+  Future<UploadResult> _uploadImage(String imageId) async {
```

Comment:
We already have `_uploadImage` in `CreatePostNotifier`, so let’s consolidate it in one location to ensure a single implementation is used.